java/com/google/flatbuffers/FlatBufferBuilder.java

```diff
@@ -527,6 +537,37 @@ public int createVectorOfTables(int[] offsets) {
         return createVectorOfTables(offsets);
     }
 
+    /**
+    * Encode the String `s` in the buffer using UTF-8. If a String with
+    * this exact contents has already been serialized using this method,
+    * instead simply returns the offset of the existing String.
+    *
+    * Usage of the method will inccur into additional allocations,
+    * so it is advisable to use it only when it is known upfront that
+    * your message will have several repeated strings.
+    *
+    * @param s The String to encode.
+    * @return The offset in the buffer where the encoded String starts.
+    */
+    public int createSharedString(String s) {
+
+        if (string_pool == null) {
+            string_pool = new HashMap<>();
+            int offset = createString(s);
+            string_pool.put(s, offset);
+            return offset;
+
+        }
+
+        Integer offset = string_pool.get(s);
```

Comment:
How does this handle buffer resizing? Would the old offsets stored in this map be invalidated when the buffer grows in size?

Reply:
AFAIk offsets are relatives and buffer resizer does not change offsets. Maybe @aardappel can correct me if I am wrong.

Also updated the comments to add a warning about additional allocations.

Reply:
Yep, i just doubled checked myself. growByteBuffer() copies to the back of the new buffer, and offsets are relative to the end. Just wanted to double check.

Reply:
Yup, most implementations use offsets from the end for exactly this reason.